lib/modem_info/modem_info_params.c

```
@@ -0,0 +1,176 @@
+
+	mcc->value = (double)strtol(mcc->string, NULL, 10);
+	mnc->value = (double)strtol(mnc->string, NULL, 10);
+
+	return 0;
+}
+
+static int cellid_to_dec(struct lte_param *cellID, double *cellID_dec)
+{
+	/* Parses the string, interpreting its content as an	*/
+	/* integral number with base 16. (Hexadecimal)		*/
+
+	*cellID_dec = (double)strtol(cellID->string, NULL, 16);
+
+	return 0;
+}
+
+static int modem_data_get(struct lte_param *param)
+{
+	enum at_param_type data_type;
+	int ret;
+
+	data_type = modem_info_type_get(param->type);
+
+	if (data_type < 0) {
+		return -EINVAL;
+	}
+
+	if (data_type == AT_PARAM_TYPE_STRING) {
+		ret = modem_info_string_get(param->type, param->string);
+
+		if (ret < 0) {
+			LOG_DBG("Link data not obtained: %d", ret);
+			return ret;
+		}
+	} else if (data_type == AT_PARAM_TYPE_NUM_SHORT) {
+		ret = modem_info_short_get(param->type, &param->value);
+
+		if (ret < 0) {
+			LOG_DBG("Link data not obtained: %d", ret);
+			return ret;
+		}
+	}
+
+	return 0;
+}
+
+int modem_info_params_get(struct modem_param_info *modem)
+{
+	int ret;
+
+	if (modem == NULL) {
+		return -EINVAL;
+	}
+
+	ret = modem_data_get(&modem->network.current_band);
+	ret += modem_data_get(&modem->network.sup_band);
+	ret += modem_data_get(&modem->network.ip_address);
+	ret += modem_data_get(&modem->network.ue_mode);
+	ret += modem_data_get(&modem->network.current_operator);
+	ret += modem_data_get(&modem->network.cellid_hex);
+	ret += modem_data_get(&modem->network.area_code);
+	ret += modem_data_get(&modem->network.lte_mode);
+	ret += modem_data_get(&modem->network.nbiot_mode);
+	ret += modem_data_get(&modem->network.gps_mode);
+
+	ret += mcc_mnc_parse(&modem->network.current_operator,
+			     &modem->network.mcc,
+			     &modem->network.mnc);
+	ret += cellid_to_dec(&modem->network.cellid_hex,
+			     &modem->network.cellid_dec);
+	ret += area_code_parse(&modem->network.area_code);
+
+	if (ret) {
+		LOG_DBG("Network data not obtained: %d", ret);
+		return ret;
```

Returning `ret` here can lead to confusion - it's not necessarily a single error code, but could be a few combined. Better to return a single error code that would indicate that parsing failed. Or prepare some macro that'd do the error check after each call.

Same below and below.

---

Fixed.